protocol/query/method/subscribe.json

```diff
@@ -1,33 +1,35 @@
 {
-	"$schema": "http://json-schema.org/draft-07/schema#",
-	"$id": "https://raw.githubusercontent.com/dedis/student_21_pop/master/protocol/query/method/subscribe.json",
-	"description": "Match subscribe to a channel client query",
-	"type": "object",
-	"properties": {
-		"method": {
-			"description": "[String] operation to be performed by the query",
-			"const": "subscribe"
-		},
-		"params": {
-			"type": "object",
-			"properties": {
-				"channel": {
-					"description": "[String] name of the channel",
-					"$ref": "channel/subChannel.json"
-				}
-			},
-			"additionalProperties": false,
-			"required": [
-				"channel"
-			]
-		},
-		"id": {
-			"type": "integer"
-		}
-	},
-	"required": [
-		"method",
-		"params",
-		"id"
-	]
-}
\ No newline at end of file
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "$id": "https://raw.githubusercontent.com/dedis/student_21_pop/master/protocol/query/method/subscribe.json",
+    "description": "Match subscribe to a channel client query",
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+        "method": {
+            "description": "[String] operation to be performed by the query",
+            "const": "subscribe"
+        },
+
+        "params": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "channel": {
+                    "description": "[String] name of the channel",
+                    "type": "string",
+                    "pattern": "^/root(/[^/]+)+$"
+                }
+            },
+
+            "required": ["channel"]
+        },
+
+        "jsonrpc": {
+            "$comment": "Defined by the parent, but needed here for the validation"
+        },
+
+        "id": {
+            "$comment": "Defined by the parent, but needed here for the validation"
```

Comment:
I see you've added these comments everywhere, but I'm pretty sure it's not necessary for validation.